### PR TITLE
addresses most zap problems

### PR DESCRIPTION
**Pull request by rlarkin212**
if we could upgrade next to 9.5 that would be nice would stop duplication https://nextjs.org/docs/api-reference/next.config.js/headers, but it works so mhhh ....

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -14,13 +12,11 @@ export default class MyApp extends App {
...
     return (
       <StateWrapper>
-        <CacheProvider value={cache}>
-          {globalStyles}
-          <ThemeProvider theme={theme}>
-            <Component {...pageProps} />
-            <ToastContainer autoClose={2000} hideProgressBar />
-          </ThemeProvider>
-        </CacheProvider>
+        {globalStyles}
```
Comment (Dragon1320) [inline pages/_app.tsx:15]: just a question, any reason why the CacheProvider thing was removed, was it doing anything before or?

Comment (rlarkin212) [reply pages/_app.tsx:15]: yeah it wasn't setting a cache at all

File: pages/_document.tsx [hunk 2]
```diff
@@ -51,6 +51,16 @@ export default class MyDocument extends Document {
   static async getInitialProps(ctx) {
     const initialProps = await Document.getInitialProps(ctx);
     const styles = extractCritical(initialProps.html);
+
+    if (ctx.pathname != "/404") {
```
[the remark this answers is not in the document]
Comment (rlarkin212) [reply pages/_document.tsx:55]: pre rendered error pages have problems setting headers as you're trying to set after the request has been sent, can't find a way around it or can't articulate the correct google search to find a way

[the remark this answers is not in the document]
Comment (rlarkin212) [reply pages/_document.tsx]: Error occurred prerendering page "/404". Read more: https://err.sh/next.js/prerender-error
TypeError: Cannot read property 'setHeader' of undefined

> Build error occurred
Error: Export encountered errors
    at _default (/Users/ryanlarkin/Workspace/winget-run/web/node_modules/next/dist/export/index.js:20:1164)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async build (/Users/ryanlarkin/Workspace/winget-run/web/node_modules/next/dist/build/index.js:36:218)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Pull request merged**